### PR TITLE
Add initial minimal MeTTa implementation

### PR DESCRIPTION
**Pull request by vsbogd**
Add minimal MeTTa interpreter implementation in Rust (`interpreter2.rs`) and MeTTa interpreter written using the minimal MeTTa instructions (`stdlib2.rs`). One can switch to the new implementation editing [`metta/runner/mod.rs` lines 13-20](https://github.com/vsbogd/hyperon-experimental/blob/c394546eec90244084cd4e82f9ac0baa1c04e11d/lib/src/metta/runner/mod.rs#L12-L20).

As implementation is not finished yet and not fully integrated with a code in this mode about 3 old Rust unit tests fails. But all unit tests in `interpreter2.rs` and `stdlib2.rs` are expected to be green. Main change is inside `...2.rs` files, other code is almost not touched.

### CODE REVIEW

**Comment by luketpeterson** [comment]
Given the scope of the change, I agree with your idea to have both coexist in main, while everyone has a chance to run them side-by-side to be sure nothing was broken.

**Comment by vsbogd** [comment]
Current version of the minimal MeTTa interpreter reproduces main aspects of the old interpreter behaviour but probably not all of them. This is one reason of failing tests and it requires fixing minimal interpreter's code to fix it.

Second reason is that change requires running interpreter manually using `!(eval (interpreter <atom> <type> <space>))` construction. This is not compatible with old interpreter behaviour and I am not sure yet how to better fix it. One way is to wrap the atom executed by user by `eval` and `interpreter` in REPL or runner. This will make old code compatible with minimal MeTTa but raises question how can we allow user to redefine the interpreter.

Bottom line I expect both code and test changes are needed to make build green and switch to the new version. But as we cannot disable tests which don't work for minimal MeTTa yet some of them will be red until minimal MeTTa becomes main version of interpreter.

**Comment by luketpeterson** [comment]
> Bottom line I expect both code and test changes are needed to make build green and switch to the new version. But as we cannot disable tests which don't work for minimal MeTTa yet some of them will be red until minimal MeTTa becomes main version of interpreter.

In my opinion the tests don't all need to pass with minimal MeTTa enabled.  It is enough that they pass with it disabled for now.  Of course all the tests will need to be updated before the default is changed and the old interpreter is removed.

**Review by luketpeterson — APPROVED** [review]
I haven't fully understood all of the code in the new interpreter, but I am happy to merge it because it is disabled by default.

File: lib/src/metta/interpreter2.rs [hunk 1]
```diff
@@ -0,0 +1,763 @@
...
+        Some([op, args @ ..]) if *op == CHAIN_SYMBOL => {
+            match args {
+                [nested, Atom::Variable(var), templ] => chain(space, bindings, nested, var, templ),
+                _ => {
+                    let error: String = format!("expected: ({} <nested> (: <var> Variable) <templ>), found: {}", CHAIN_SYMBOL, atom);
+                    vec![InterpretedAtom(error_atom(atom, error), bindings)]
+                },
+            }
+        },
+        Some([op, args @ ..]) if *op == MATCH_SYMBOL => {
```
[the remark this answers is not in the document]
Comment (vsbogd) [reply lib/src/metta/interpreter2.rs:220]: Yes, we can implicitly use `Empty` if `else` is not used, thank you for the idea.

**Review by Necr0x0Der — APPROVED** [review]
